jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssTravelerMessageBuilder.java

```diff
@@ -676,19 +679,22 @@ public static HeadingSlice getHeadingSlice(String heading) {
       }
    }
    
-   public static MsgCRC getMsgCrc(String sum) {
-      if (sum == null || sum.length() == 0) {
-         return new MsgCRC(new byte[] { 0X00, 0X00 });
+

```

Comment:
I can envision that we will need to do similar conversions in other places. Therefore, it would be beneficial to create a utility for this type of conversion. Ideally, the utility method should be able to convert a string representation of a byte array of any length. But for now, we can restrict it to a short integer, hence the length can be either 16 or 4 as you have it. To that end, I recommend that we move the following logic to CodecUtils in jpo-ode-common as follows:

```
    /**
     * @param strShort String representation of a short integer value in binary or hex format.
     *                  If the string is in binary format, the length must be exactly 16 1s and zeros.
     *                  If Hex format, the length must be exactly 4 Hex digits.
     * 
     * @return a byte array equivalent of strShort 
     */
    public static byte[] shortStringToByteArray(String strShort) {
        
        byte[] byteArrayValue = null;
        
        int radix = radixOf(strShort);

        if (radix == 0) {
            byteArrayValue = new byte[2]; //NOSONAR
        } else {
            byteArrayValue = 
                    Arrays.copyOfRange(ByteBuffer.allocate(4).putInt(
                        Integer.parseUnsignedInt(strShort, radix)).array(), 2, 4);
        }

        return byteArrayValue;
     }

    /**
     * @param strShort String representation of a short integer value in binary or hex format.
     *                  If strShort is in binary format, the length must be exactly 16 ones and zeros.
     *                  If strShort is in Hex format, the length must be exactly 4 Hex digits.
     * @return The radix of the strShort: Currently supporting only binary and hex, therefore
     *         the return value is either 2 or 16 
     */
    private static int radixOf(String strShort) {
        int radix = 0;
        if (strShort == null || strShort.length() == 0) {
           radix = 0;
        } else if (strShort.length() == 16) {
           radix = 2;
        } else if (strShort.length() == 4) {
           radix = 16;
        } else {
           throw new IllegalArgumentException("Short String length is invalid: " + strShort.length());
        }
        return radix;
    }
```
And use the util in OssTravelerMessageBuilder#getMsgCrc():

```
   public static MsgCRC getMsgCrc(String sum) {
      return new MsgCRC(CodecUtils.shortStringToByteArray(sum));
   }
```

As you can see, I removed the call DatatypeConverter.parseHexBinary(msgString) in lieu of using the same technique for both binary and Hex. I have not tested it but if it works, I think it's better to use one method to avoid running into inconsistencies cause by different implementations of the JVM.